tests/unit/map_test.c

```diff
@@ -516,6 +516,154 @@ static void test_hashmap_degenerate_hash_fn(void)
     HashMapDestroy(hashmap);
 }
 
+#define ARRAY_STRING_KEY(last_char) {'k', 'e', 'y', last_char, '\0'}
+#define ARRAY_STRING_VALUE(last_char) {'v', 'a', 'l', 'u', 'e', last_char, '\0'}
+
+/**
+ * @brief Return pointer to heap allocated string constructed as
+ *        "keyX" where X is last_char.
+ */
+static char *MallocStringKey(char last_char)
+{
+    return xstrdup((char[]) ARRAY_STRING_KEY(last_char));
+}
+
+/**
+ * @brief Return pointer to heap allocated string constructed as
+ *        "valueX" where X is last_char.
+ */
+static char *MallocStringValue(char last_char)
+{
+    return xstrdup((char[]) ARRAY_STRING_VALUE(last_char));
+}
+
+static void test_array_map_insert(void)
+{
+    ArrayMap *m = ArrayMapNew(StringEqual_untyped, free, free);
+
+    // Test simple insertion
+    for (int i = 0; i < 3; ++i)
+    {
+        char* k = MallocStringKey('A' + i);
+        char* v = MallocStringValue('A' + i);
+        assert_int_equal(ArrayMapInsert(m, k, v), 2);
+    }  // size == 3
```

Comment:
Could have been an `assert()`.

Reply:
indeed, i will use assert_int_equals though because just assert() didnt perform as expected. Wherever i use assert_true i used asser() first and it dint work properly.